python/python/tests/test_commit_index.py

```diff
@@ -56,13 +58,22 @@ def test_commit_index(dataset_with_index, test_table, tmp_path):
 
     # Commit the index to dataset_without_index
     field_idx = dataset_without_index.schema.get_field_index("meta")
```

Comment:
FYI this is wrong, we want the field id, not the field index. They are often the same, but not always (especially if you have nested data). Not something we need to fix now, but what you to be aware of this.